### PR TITLE
Update dependencies 🦆

### PR DESCRIPTION
**Pull request by MohamedBsh**
@Bl3f 

This PR updates several key dependencies in our pyproject.toml file to improve compatibility with Python 3.12 and resolve installation issues, particularly with DuckDB. It also adjusts the Python version range to ensure compatibility with pandas.

**_Changes_**

Upgraded duckdb from ^0.9.2 to ^1.0.0
Upgraded sqlglot from ^22.0.1 to ^23.0.2
Upgraded pandas from 2.0.3 to ^2.2.0
Changed Python version range from >=3.8.1,<3.13 to >=3.9,<3.13

**_Rationale_**

These upgrades aim to resolve installation issues encountered when using Poetry with Python 3.12 #4 , specifically a ChefBuildError related to DuckDB installation. The newer versions of these dependencies offer better compatibility with Python 3.12 and include various improvements and bug fixes.
The Python version range was adjusted to >=3.9,<3.13 to accommodate pandas 2.2.0, which requires Python 3.9 or later.

**_Testing_**

```
(base) poetry update       
Updating dependencies
Resolving dependencies... (1.9s)

Package operations: 23 installs, 0 updates, 0 removals

  - Installing six (1.16.0)
  - Installing jmespath (1.0.1)
  - Installing python-dateutil (2.9.0.post0)
  - Installing urllib3 (2.2.2)
  - Installing botocore (1.35.1)
  - Installing mdurl (0.1.2)
  - Installing click (8.1.7)
  - Installing iniconfig (2.0.0)
  - Installing markdown-it-py (3.0.0)
  - Installing mypy-extensions (1.0.0)
  - Installing packaging (24.1)
  - Installing pathspec (0.12.1)
  - Installing platformdirs (4.2.2)
  - Installing pluggy (1.5.0)
  - Installing pygments (2.18.0)
  - Installing s3transfer (0.10.2)
  - Installing black (24.8.0)
  - Installing boto3 (1.35.1)
  - Installing duckdb (0.10.3)
  - Installing isort (5.13.2)
  - Installing pytest (8.3.2)
  - Installing rich (13.7.1)
  - Installing sqlglot (23.17.0)
```

**_Notes_**

The Python version range has been narrowed to >=3.9,<3.13 to ensure compatibility with pandas 2.2.0.
This change drops support for Python 3.8. If support for 3.8 is crucial, we may need to consider using an older version of pandas or creating separate dependency sets for different Python versions.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,12 +23,12 @@ packages = [
 yato = 'yato.cli:cli'
 
 [tool.poetry.dependencies]
-python = ">=3.8.1,<3.13"
+python = ">=3.9,<3.13"
 boto3 = "^1.34.51"
-sqlglot = "^22.0.1"
-duckdb = "^0.9.2"
+sqlglot = "^23.0.2"
+duckdb = "^0.10.0"
```
Comment (jbondu9) [inline pyproject.toml]: Hi @MohamedBsh, a new version has arrived! Very useful for using Yato with MotherDuck and for developers on Windows 🙂 
```suggestion
duckdb = "^1.0.0"
```

Comment (MohamedBsh) [reply pyproject.toml]: Thank you @jbondu9 . Your suggestion is relevant and has been implemented. DuckDB has been updated to version 1.0.0 in the pyproject.toml file 🤗

**Comment by benjaminsicard** [comment]
I confirm the need to merge this PR.

I tried to use yato for the first time today, and here is my experience:
- first I installed yato in a venv with python 3.12 --> not working because of duckdb dependancies
- then I used a python 3.8 venv --> not working because it could not import 'TopologicalSorter' from the 'graphlib' library
- then I used a python 3.9 venv --> this time I could install but I could not run yato because of a duckdb internal error (Unsupported compression function type)
- finally, I forked the repo, implemented the changes from this PR, installed my own yato, and everything worked ;-)

**Comment by Bl3f** [comment]
Hey folks, thank you so much for your contributions and tries. I've merged @MohamedBsh PR to bump everything. 

I publish a new version on PyPi during the day! So happy to have you here, I'm gonna get back yato on rails very soon!

**Comment by benjaminsicard** [comment]
> Hey folks, thank you so much for your contributions and tries. I've merged @MohamedBsh PR to bump everything.
> 
> I publish a new version on PyPi during the day! So happy to have you here, I'm gonna get back yato on rails very soon!

Thank you for your quick resolution @Bl3f ! Has it something to do with nao :-D ? Very excited for this by the way.

Also, at the moment, my experience with yato is pretty good, very useful for small adhoc tasks on csv files etc. that needs multiple transformations.
As a Data Engineer, sometimes I need to get my hands dirty and I found duckdb very relevant to handle csv files. However, when I have a lot of transformation steps to perform (cleaning, joining, aggregating etc.), I find it difficult to only use one big sql script to operate my DB. I need to split it in several files for better readability. 

At first I added a new profile to our dbt-core project, but it meant that I also had to:
-  install duckdb adapter (sometimes it can conflict with an existing adapter, I had some bad experiences with dbt-postgres and dbt-athena haha)
- update the dbt project yml
- declare sources
- etc.

I stumbled on one article from Julien Hurault (https://juhache.substack.com/p/pip-install-data-stack) that was mentioning yato as a 'lightweight dbt'.
Now for the actual adhoc data project I need to work on in my job, it's far easier to use yato by dumping my sql transformation files in the /sql folder ;-)

**Comment by Bl3f** [comment]
Yeah, actually I use a few internal functions that I've developed in yato inside nao 🙈.

Thank you so much for your feedback, when I read this I see exactly the why I developed yato, which is the simplest way to run SQL queries without the burden of a dbt setup.

Once you have a few days of usage that would be interesting to get your feedback on what should we add next to yato (without adding more complexity) I think @MohamedBsh wants to do some stuff :).

**Comment by MohamedBsh** [comment]
@benjaminsicard  Don't hesitate to open issues or feature requests for the project if you have any suggestions. If you'd like to discuss further, feel free to reach out to me via email or we could set up a call.
For more information about me or to get in touch, you can check out my profile [here](https://mohamedaminebousahih.notion.site/Hi-I-m-Mohamed-Amine-Bousahih-5e6e0beb16004939a356a1cdcc9530ad)
I look forward to potentially collaborating on improving yato!

**Pull request merged**